### PR TITLE
dynamically add env vars for custom installer build into docker env

### PR DESCRIPTION
**Pull request by lyw07**
### Summary

This PR is to add a script to dynamically add env vars prefixed with KOLIBRI_BUILD_ into docker environment for building the UN Women custom installers.
The script `build_tools/customize_docker_envlist.py` will parse through all the environment variables in the current environment, find all the variables prefixed with `KOLIBRI_BUILD_`, remove the prefix and pass the result name and corresponding value to `env.list` so that docker can read from this file.
The env.list will be restored after running docker container so that the dynamic env vars will not be committed to the repo.

### Reviewer guidance

testing buildkite pipeline:
1. `export KOLIBRI_BUILD_XXX=a`
2. run `make docker-whl` 
3. run `docker exec -it <kolibri_container>` to see if the environment variable `XXX` exists

### References

#5769 

----

### Contributor Checklist

PR process:

- [X] PR has the correct target branch and milestone
- [X] PR has 'needs review' or 'work-in-progress' label
- [X] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [X] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: docker/build_whl.dockerfile [hunk 1]
```diff
@@ -45,8 +45,8 @@ COPY . .
 
 CMD echo '--- Installing JS dependencies' && \
     yarn install --pure-lockfile && \
-    echo `--- Making whl` && \
```
Comment (lyw07) [inline docker/build_whl.dockerfile:48]: This is just to fix the error `/bin/sh: 1: ---: not found` when running the dockerfile

File: build_tools/__init__.py [hunk 2]
```diff
@@ -32,4 +35,6 @@
 For BUILD_TIME_PLUGINS, RUN_TIME_PLUGINS, and EXTRA_REQUIREMENTS if the environment variable
 specifies a URL, the scripts will attempt to download a text file from that URL and then use
 that file.
+
+BACKGROUND_IMAGE, an image to display on the Kolibri login page.
```
Comment (rtibbles) [inline build_tools/__init__.py]: Should remove this, as it is specific to the UN women plugin.

Comment (lyw07) [reply build_tools/__init__.py]: thanks! I will remove it!

File: docker/env.list [hunk 3]
```diff
@@ -29,5 +25,3 @@ KOLIBRI_RUN_MODE="docker"
 # DOCKERMNT_PEX_PATH=
 # Specify path to a .pex file in the /docker/mnt volume, e.g.
 # DOCKERMNT_PEX_PATH=/docker/mnt/kolibri-0.11.0a1.dev_git-37-gb15b923.pex
```
Comment (rtibbles) [inline docker/env.list:27]: There doesn't seem to be a new line at the end of this file, so I wonder if we need to make sure that a new line is added before adding any env vars in the python script above.

File: build_tools/customize_docker_envlist.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    envs = os.environ
+
+    with open(ENVLIST_FILE, "a") as f:
+        for env in envs:
+            # Find all the environment variables prefixed with KOLIBRI_BUILD_
+            # and add them to env.list to pass into docker environment after
+            # removing the prefix.
+            if env.startswith(BUILD_ENV_PREFIX):
+                key = env.replace(BUILD_ENV_PREFIX, "")
+                f.write("{key}={value}\n".format(key=key, value=envs[env]))
```
Comment (rtibbles) [inline build_tools/customize_docker_envlist.py]: Per my comment below, it's probably safer to prefix the write with the newline, to make sure we don't accidentally add to an existing line. So:
`f.write("\n{key}={value}".format(key=key, value=envs[env]))`

**Review by rtibbles — CHANGES_REQUESTED** [review]
2 small changes, but the overall implementation looks right!

File: build_tools/customize_docker_envlist.py (hunk 4 above)
Comment (lyw07) [reply build_tools/customize_docker_envlist.py]: makes sense to me. I will make the change soon!

**Review by rtibbles — APPROVED** [review]

**Pull request merged**